crates/ott-balancer/src/balancer.rs

```diff
@@ -171,12 +171,95 @@ impl BalancerLink {
     }
 }
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct BalancerContext {
     pub clients: HashMap<ClientId, BalancerClient>,
     pub monoliths: HashMap<MonolithId, BalancerMonolith>,
     pub rooms_to_monoliths: HashMap<RoomName, RoomLocator>,
     pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
+    pub monolith_selection: Box<dyn MonolithSelection + Send + Sync + 'static>,
+}
+#[derive(Debug, Default)]
+pub struct MonolithRegistry;
+pub trait MonolithSelection: std::fmt::Debug {
+    fn select_monolith(
+        &self,
+        monolith: Vec<&BalancerMonolith>,
+        ctx: &BalancerContext,
```

Comment:
don't take `ctx` as an argument, that tightly couples the trait to `BalancerContext` and will make it harder to write unit tests